modules/navier_stokes/include/actions/AddNavierStokesVariablesAction.h

```diff
@@ -0,0 +1,43 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef ADDNAVIERSTOKESVARIABLESACTION_H
+#define ADDNAVIERSTOKESVARIABLESACTION_H
+
+#include "NSAction.h"
+
+class AddNavierStokesVariablesAction;
+
+template<>
+InputParameters validParams<AddNavierStokesVariablesAction>();
+
+/**
+ * This class allows us to have a section of the input file like the
+ * following which automatically adds all the required nonlinear
+ * variables with the appropriate scaling.
+ *
+ * [NavierStokes]
```

Comment:
I agree that this should be the default, but in things like bighorn, you want that to be a top-level block, since it is the core of your input file. During the tiger team, I'd like to work on that custom prefix, so that apps can register these things according to their needs. This feature would be used in RELAP for the fluid properties system. Bighorn will probably do that same.

I can seem to find a ticket for the custom prefix...